src/test/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorTest.java

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.meta;
+
+import static org.junit.Assert.assertEquals;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.internal.utils.CheckUtil;
+
+public final class MetadataGeneratorTest {
+    @Test
+    public void testBasic() throws Exception {
+        MetadataGenerator.generate(System.getProperty("user.dir")
+                + "/src/main/java/com/puppycrawl/tools/checkstyle");
+        final Set<String> metaFiles;
+
+        try (Stream<Path> fileStream = Files.walk(
+                Paths.get(System.getProperty("user.dir") + "/src/main/resources/com/puppycrawl"
+                        + "/tools/checkstyle/meta"))) {
+            metaFiles = fileStream
+                    .filter(Files::isRegularFile)
+                    .map(file -> {
+                        final String fileName = file.getFileName().toString();
+                        final int lengthToOmit;
+                        if (fileName.contains("Check")) {
+                            lengthToOmit = "Check.xml".length();
+                        }
+                        else {
+                            lengthToOmit = ".xml".length();
+                        }
+                        return fileName.substring(0, fileName.length() - lengthToOmit);
+                    })
+                    .sorted()
+                    .collect(Collectors.toCollection(LinkedHashSet::new));
+        }
+        final Set<String> checkstyleModules =
+                CheckUtil.getSimpleNames(CheckUtil.getCheckstyleModules())
+                .stream()
+                .sorted()
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+        checkstyleModules.removeAll(Arrays.asList("Checker", "TreeWalker",
+                "JavadocMetadataScraper"));
```

Comment:
`Arrays.asList("Checker", "TreeWalker", JavadocMetadataScraper")`
this list can be extracted as field

Reply:
Done